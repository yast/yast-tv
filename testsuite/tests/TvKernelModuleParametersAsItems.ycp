{
    // testedfiles: Tv.ycp

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    define void TEST_It_Now(string s_module, map params) ``{
	DUMP("---------------");

	Tv::kernel_modules = nil;
	list items = Tv::KernelModuleParametersAsItems(s_module,params);
	DUMP(items);


	Tv::kernel_modules = $[];
	items = Tv::KernelModuleParametersAsItems(s_module, params);
	DUMP(items);


	Tv::kernel_modules = $[
		"bttv":$["module_description":"bttv - v4l driver module for bt848/878 based cards", "tuner":"specify installed tuner type"],
	];
	items = Tv::KernelModuleParametersAsItems(s_module, params);
	DUMP(items);

    }

    TEST_It_Now("bttv", $[ "non_existing_parameter" : "its_value" ]);

	TEST_It_Now("bttv", $[ "tuner" : "1" ]);
}
