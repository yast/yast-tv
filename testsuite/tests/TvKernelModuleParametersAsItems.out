Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.Instmode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	---------------
Dump	[]
Dump	[]
Dump	[`item (`id ("tuner"), "tuner", "", "specify installed tuner type")]
Dump	---------------
Dump	[]
Dump	[]
Dump	[`item (`id ("tuner"), "tuner", "1", "specify installed tuner type")]
